### DISCUSSION TITLE
[issue] Filter Local/Private IP's

### DISCUSSION THREAD

**Discussion by rtcms** [projectdiscovery/cloudlist#15] [Q&A] [answered]
**Describe the bug**
`cloudlist -ip -silent` gives me private IP's as well (I am not sure if this is inteneded)

**Nuclei version**
Please share the version of the nuclei you are running with `dnsx -version` 
`[INF] Current Version: 1.0.3
`

**Screenshot of the error or bug**
![Screenshot from 2021-05-30 18-51-56](https://user-images.githubusercontent.com/35173068/120105821-35560c00-c178-11eb-88f2-c9f3e541e472.png)

**Expected Solution**
This tool should essentially filter out all the [Reserved IP's ](https://en.wikipedia.org/wiki/Reserved_IP_addresses) as mentioned by IANA

**Comment by ehsandeep** [comment]
Hi @rtcmsm, 

This is expected, idea is to list all the cloud assets in one place, and they will be auto filtered in the next step, i.e when running httpx or anything similar, but removing them as default will cut down few use cases where you required all the data.

**Reply by rtcms** [reply]
Definitely agree with that. It will cut down some IP's when performing internal pen-testing or similar activity.
Can we have a explicit flag like `--exclude-IANA `or something like that(not enabled by default)?

**Reply by ehsandeep** [reply, marked as answer]
Sure, adding flag support will be good idea, GH issue - https://github.com/projectdiscovery/cloudlist/issues/16

**Reply by rtcms** [reply]
Thanks :)
